device/iot-device-samples/pnp-device-sample/temperature-controller-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/PnpConvention.java

```diff
@@ -92,18 +105,39 @@ public static Message createIotHubMessageUtf8(@NonNull String telemetryName, @No
      *     }
      * }
      */
-    public static Set<Property> createPropertyPatch(@NonNull final String propertyName, @NonNull final Object propertyValue, String componentName) {
+    public static Set<Property> createComponentPropertyPatch(@NonNull final String propertyName, @NonNull final Object propertyValue, @NonNull String componentName) {
+
+        return createComponentPropertyPatch(componentName, new HashMap<String, Object>()
+        {{
+            put(propertyName, propertyValue);
+        }});
+    }
+
+    /**
+     * Create a key-value property patch for both read-only and read-write properties.
+     * @param propertyKeyValuePairs The property name and an unserialized value, as defined in the DTDL interface.
+     * @param componentName (optional) The name of the component in which the property is defined. Can be null for property defined under the root interface.
```

Comment:
this says componentName is optional, and can be null. Yet the argument is decorated with "NonNull". Doesn't that make componentName required?

Reply:
Good observation, will modify comment